chalk-solve/src/solve/slg/unique.rs

```diff
@@ -0,0 +1,17 @@
+use crate::solve::slg::SlgContext;
+use crate::solve::slg::SlgContextOps;
+use chalk_ir::*;
+
+use chalk_engine::context;
+use chalk_engine::SimplifiedAnswer;
+
+/// Takes one answer and returns it as unique solution
+impl<'me> context::UniqueOps<SlgContext> for SlgContextOps<'me> {
+    fn make_unique_solution(
```

Comment:
I don't really understand this name. It doesn't seem like it's making a "unique" solution -- it seems like it's just extracting the canonical substitution from the answer? Maybe we should just call it `constrained_subst_from_answer`  or something?

I think I would also just add it to `ContextOps` instead of creating a new trait.

Reply:
I've moved and renamed this function.